CONTRIBUTING.md

```diff
@@ -0,0 +1,257 @@
+
+### Logs
+
+We use SLF4J; loggers are declared like this:
+
+```java
+private static final Logger LOG = LoggerFactory.getLogger(TheEnclosingClass.class);
+```
+
+
+### Nullability annotations
+
+We use the [Spotbugs annotations](https://spotbugs.github.io) to document nullability of parameters,
+method return types and class members.
+
+Please annotate any new class or interface with the appropriate annotations: `@NonNull`, `@Nullable`. Make sure you import 
+the types from `edu.umd.cs.findbugs.annotations`, there are homonyms in the classpath.
+
+
+## Coding style -- test code
+
+Static imports are permitted in a couple of places:
+* All AssertJ methods, e.g.:
+  ```java
+  assertThat(node.getDatacenter()).isNotNull();
+  fail("Expecting IllegalStateException to be thrown");
+  ```
+* All Mockito methods, e.g.:
+  ```java
+  when(codecRegistry.codecFor(DataTypes.INT)).thenReturn(codec);
+  verify(codec).decodePrimitive(any(ByteBuffer.class), eq(ProtocolVersion.DEFAULT));
+  ```
+
+Test methods names use lower snake case, generally start with `should`, and clearly indicate the
+purpose of the test, for example: `should_fail_if_key_already_exists`. If you have trouble coming 
+up with a simple name, it might be a sign that your test does too much, and should be split.
+
+We use AssertJ (`assertThat`) for assertions. Don't use JUnit assertions (`assertEquals`, 
+`assertNull`, etc).
+
+Don't try to generify at all cost: a bit of duplication is acceptable, if that helps keep the tests
+simple to understand (a newcomer should be able to understand how to fix a failing test without
+having to read too much code).
+
+Test classes can be a bit longer, since they often enumerate similar test cases. You can also
+factor some common code in a parent abstract class named with "XxxTestBase", and then split
+different families of tests into separate child classes.
+
+## Running the tests
+
+### Unit tests
+
+    mvn clean test
+   
+### Integration tests
+
+    mvn clean verify
+
+### Integration tests native
+
+    mvn clean verify -Dnative
+    
+### Generating documentation
+   
+    mvn clean package -Prelease    
+    
+
+## Build
+
+* Clone the repository: `git clone git@github.com:datastax/cassandra-quarkus.git`
+* Navigate to the directory: `cd cassandra-quarkus`
+* Invoke `mvn clean install -DskipTests=true -Dmaven.javadoc.skip=true -B -V` from the root directory
+
+```bash
+git clone git@github.com:datastax/cassandra-quarkus.git
+cd cassandra-quarksu
+mvn clean install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
+```
+
+This build skipped all the tests, native-image builds and documentation generation. 
+
+Removing the `-DskipTests -DskipITs` flags enables the tests. 
+It will take much longer to build but will give you more guarantees on your code. 
+
+You can build and test native images in the integration tests supporting it by using `mvn install -Dnative`.
+
```

Comment:
We might want to add more sections:

# Continuous Integration (CI)

All branches and pull-requests in this project are built regularly on DataStax internal continuous integration servers. These builds are not publicly available at this moment.

The `master` branch of this project is also regularly built against Quarkus `master` branch, by Quarkus own CI system. The builds can be consulted [here](https://github.com/datastax/cassandra-quarkus/actions?query=workflow%3A%22Quarkus+ecosystem+CI%22).

# Deployment & Release

DataStax does not publish snapshot builds of this project at the moment.

Releases are managed and conducted by DataStax. Stable (release) artifacts are available [from Maven Central](https://repo1.maven.org/maven2/com/datastax/oss/quarkus/cassandra-quarkus-parent). See "Getting the extension" in the project's main [README file](https://github.com/datastax/cassandra-quarkus/#getting-the-extension) for more details.